### ISSUE TITLE
"IO error - EOF" error after calling producer.Stop()

### ISSUE THREAD

**Issue by tmpapageorgiou** [nsqio/go-nsq#212]
The error message are generate intermitantly lways after call producer.Stop()

`(localhost:4150) IO error - EOF`

The following code reproduces the problem:

``` go
package main                                                                                                                        
                                                                                                                                    
import (                                                                                                                            
    "bitbucket.org/tpapageorgiou/mytest/broker"                                                                   
    log "github.com/Sirupsen/logrus"                                                                                                
    "github.com/nsqio/go-nsq"                                                                                                       
)                                                                                                                                   
                                                                                                                                    
func main() {                                                                                                                       
    config := nsq.NewConfig()                                                                                                       
                                                                                                                                    
    buffer := []byte{`{"test": "123"}`}                                                                                                            
                                                                                                                                    
    for i := 0; i < 100; i++ {                                                                                                      
        producer, _ := nsq.NewProducer("localhost:4150", config)                                                                    
        producer.SetLogger(broker.NewNSQLogrusLoggerAtLevel(log.DebugLevel))                                                        
        err := producer.Publish("test_topic", buffer)                                                                               
                                                                                                                                    
        if err != nil {                                                                                                             
            log.Error(err)                                                                                                          
        }                                                                                                                           
        producer.Stop()                                                                                                             
    }                                                                                                                               
}   
```                                                                                                                                    
I am using:

- Ubuntu Xenial 16.04
- Go v1.8
- go-nsq v1.06
- Logrus v1.0
- [NewNSQLogrusLoggerAtLevel is a function from a helper to integrate logrus with gonsq. ](https://gist.github.com/michaeltrobinson/4d521cb4267bcf7bda9b0c775df4a440)

**Comment by tmpapageorgiou** [comment]
Thi is a diplicated of https://github.com/nsqio/go-nsq/issues/184

**Comment by ploxiln** [comment]
This is indeed a duplicate, but it was thought to have been fixed in #201 which was included in go-nsq v1.0.6
